File: Ese-Rental-Service-Frontend/src/api/authService.js
import { axiosInstance } from './axiosInstance';

export const authService = {
    login: (credentials) => axiosInstance.post('/auth/login', credentials),
    register: (userData) => axiosInstance.post('/auth/register', userData),
    forgotPassword: (request) => axiosInstance.post('/auth/forgot-password', request),
    resetPassword: (request) => axiosInstance.post('/auth/reset-password', request),
    verifyOTP: (request) => axiosInstance.post('/auth/verify-otp', request),
    logout: () => axiosInstance.post('/auth/logout'),
};

File: Ese-Rental-Service-Backend/src/main/java/com/bikalp/eserentalservice/service/AuthService.java
package com.bikalp.eserentalservice.service;

import com.bikalp.eserentalservice.dto.auth.*;
import com.bikalp.eserentalservice.entity.User;

public interface AuthService {
    /**
     * Register a new user
     * @param request the registration request
     * @return authentication response with token
     */
    AuthResponse register(RegisterRequest request);

    /**
     * Authenticate a user
     * @param request the login request
     * @return authentication response with token
     */
    AuthResponse login(LoginRequest request);

    ForgetPasswordResponse forgotPassword(ForgotPasswordRequest request);

    VerifyOTPResponse verifyOTP(VerifyOTPRequest request);

    ResetPasswordResponse resetPassword(ResetPasswordRequest request);

    void logout();
} 